.gitlab-ci.yml

```diff
@@ -0,0 +1,45 @@
+stages:
+  - test
+
+.test:
+  image: node:12
+  stage: test
+  before_script:
+    # install packages
+    - apt-get update
+    - apt-get install -y apt-transport-https ca-certificates wget dirmngr gnupg software-properties-common
+    - wget -qO - https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public | apt-key add -
+    - add-apt-repository --yes https://adoptopenjdk.jfrog.io/adoptopenjdk/deb/
+    - apt update
+    - apt install -y adoptopenjdk-8-hotspot
+    # install maven
+    - wget https://downloads.apache.org/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz -P /tmp
+    - tar xf /tmp/apache-maven-*.tar.gz -C /tmp
+    - ln -s /tmp/apache-maven-3.6.3/bin/mvn /usr/bin/mvn
+    # install protoc-gen
+    - curl -L https://github.com/grpc/grpc-web/releases/download/1.0.7/protoc-gen-grpc-web-1.0.7-linux-x86_64 --output protoc-gen-grpc-web
+    - mv protoc-gen-grpc-web /usr/local/bin/
+    - chmod +x /usr/local/bin/protoc-gen-grpc-web
+    # install npm deps
+    - npm i --cache .npm --prefer-offline --no-audit --progress=false nyc codecov protoc grpc-tools
+    - npm i --cache .npm --prefer-offline --no-audit --progress=false
+    - git submodule update --init --recursive
+
+compile typescript:
+  stage: test
+  extends: .test
+  script:
+    - npm run build
+
+run tests:
+  stage: test
+  extends: .test
+  script:
+    - ./node_modules/nyc/bin/nyc.js npm test
+
+code coverage:
+  stage: test
+  extends: .test
+  script:
+    - mkdir .nyc_output
+    - ./node_modules/nyc/bin/nyc.js report --reporter=text-lcov
```

Comment:
Is there a specific aversion to using CodeCov? Is there an alternative in GitLab CI?